ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -229,6 +229,111 @@ def test_delete_with_nonexistent_hostgroup(self, automember_hostgroup,
                 reason=u'%s: Automember rule not found' % hostgroup1.cn)):
             command()
 
+    def test_create_with_existent_group(self, automember_group, group1):
+        """ Try to add a rule that already exists """
+        group1.ensure_exists()
+        automember_group.ensure_exists()
+        command = automember_group.make_create_command()
+        with raises_exact(errors.DuplicateEntry(
+                reason=u'Automember rule with name "%s" '
+                       u'already exists' % group1.cn)):
+            command()
+
+    def test_create_with_existent_hostgroup(self, automember_hostgroup,
```

Comment:
This test is basically a copy-paste of the previous one, with different parameters. You can write a single test and parametrize it with:
```
   @pytest.mark.parametrize(
        "automembergrp, grp", [
            (automember_group, group1),
            (automember_hostgroup, hostgroup1)
        ])
    def test_create_with_existent_group(self, automember_grp, grp):
        """ Try to add a rule that already exists """
        grp.ensure_exists()
        automember_grp.ensure_exists()
        ...
```

This comment can be applied to most of your new test cases and avoids code duplication.